src/snowflake/snowpark/functions.py

```diff
@@ -9852,5 +9882,9 @@ def make_interval(
             milliseconds,
             microseconds,
             nanoseconds,
-        )
+        ),
+        _emit_ast=False,
     )
+
+    res._ast = ast
```

Comment:
this here should allow you to remove the check in `select`.

Reply:
ooh yeah, I found an issue with the test. Thanks for pointing this out!!